spec/helpers/scheduler.rb

```diff
@@ -0,0 +1,241 @@
+# frozen_string_literal: true
+
+# This file is copied from https://github.com/ruby/ruby/blob/5e9598baea97c53757f12713bacc7f19f315c846/test/fiber/scheduler.rb
+
+# This is an example and simplified scheduler for test purposes.
+# It is not efficient for a large number of file descriptors as it uses IO.select().
+# Production Fiber schedulers should use epoll/kqueue/etc.
+
+require 'fiber'
+require 'socket'
+
+begin
+	require 'io/nonblock'
+rescue LoadError
+	# Ignore.
+end
+
+module Helpers
+class Scheduler
```

Comment:
Matz wanted to incorporate Async into Ruby as a default gem but I said no. I want everyone to have equal opportunity to implement the scheduler. Maybe it was a bad choice but I want to encourage other engineers (including yourself) to create their own schedulers.

We could try to make a reference design but I think it would create soft-engineering challenges, including encouraging people to use it in a way it would not be suitable, also ossifying a particular implementation into Ruby core.